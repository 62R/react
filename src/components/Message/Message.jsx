import style from './Message.module.scss'

export const Message = ({ msg }) =>
    <p className={style.message}>
        Это сообщение было передано из App: <span className={style.message__span}> {msg} </span>
    </p>
