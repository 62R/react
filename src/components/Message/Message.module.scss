.message {
    display: flex;
    flex-direction: column;
    align-self: flex-start;
    

    &__author {
        color: darkblue;
        font-weight: bold;
        font-size: 14px;
        margin: 0 5px;
    }

    &__message {        
        overflow-wrap: break-word;
        color: black;
        font-style: italic;
        
        padding: 2px 10px 3px;
        border: 1px solid black;
        border-radius: 15px;
        width: fit-content;
    }

    &_author_bot {
        align-self: flex-end;
        align-items: end;
        text-align: right;
    }
}