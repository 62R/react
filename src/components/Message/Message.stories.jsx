import React from 'react';

import { Message } from './Message';

export default {
  title: 'MyApp/Message',
  component: Message,
};

const Template = (args) => <Message {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  msg: { author: 'John', msg: 'Some message' },
};

export const Secondary = Template.bind({});

Secondary.args = {
  msg: { author: 'Bot', msg: 'Another message' },
};
