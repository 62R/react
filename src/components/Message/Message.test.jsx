import { Message } from './Message';
import { render } from '@testing-library/react';

describe('Message', () => {
  it('render component with snaphshot', () => {
    const msg = {
      author: 'One',
      msg: 'Hello world',
    };
    const { asFragment } = render(<Message msg={msg} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
