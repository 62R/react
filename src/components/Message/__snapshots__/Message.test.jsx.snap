// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message render component with snaphshot 1`] = `
<DocumentFragment>
  <div
    class="message "
  >
    <p
      class="message__author"
    >
      One
    </p>
    <p
      class="message__message"
    >
      Hello world
    </p>
  </div>
</DocumentFragment>
`;
